gui/wxpython/gui_core/pyedit.py

```diff
@@ -400,6 +412,28 @@ def OnOpen(self, event):
         """Handle open event but ask about replacing content first"""
         if self.CanReplaceContent('file'):
             self.Open()
+            self.recent_files.AddFileToHistory(
+                filename=self.filename,
+            )
+
+    def OpenRecentFile(self, path, file_exists):
+        """Try open recent file and read content
+
+        :param str path: file path
+        :param bool file_exists: file path exists
+
+        :return: None
+        """
+        if self.CanReplaceContent(by_message='file'):
+            if not file_exists:
```

Comment:
Let's do this first. Why bother user with replacing content when the files does not exist?

Reply:
All right, I fixed it.